diff -ur TPM2.0-TSS-1.0/Makefile.am TPM2.0-TSS-1.0-new/Makefile.am
--- TPM2.0-TSS-1.0/Makefile.am	2016-11-01 06:53:15.000000000 -0700
+++ TPM2.0-TSS-1.0-new/Makefile.am	2017-03-11 16:51:38.164636194 -0700
@@ -34,7 +34,7 @@
 
 # stuff to build, what that stuff is, and where/if to install said stuff
 sbin_PROGRAMS   = $(resourcemgr)
-noinst_PROGRAMS = $(tpmclient) $(tpmtest)
+bin_PROGRAMS = $(tpmclient) $(tpmtest)
 lib_LTLIBRARIES = $(libsapi) $(libtcti_device) $(libtcti_socket)
 
 # unit tests
